stackit/internal/services/serverbackup/schedule/resource.go

```diff
@@ -0,0 +1,539 @@
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing credential",
+			fmt.Sprintf("Expected import identifier with format [project_id],[server_id],[backup_schedule_id], got %q", req.ID),
+		)
+		return
+	}
+
+	intId, err := strconv.ParseInt(idParts[2], 10, 64)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing credential",
+			fmt.Sprintf("Expected backup_schedule_id to be int64, got %q", idParts[2]),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), idParts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("server_id"), idParts[1])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("backup_schedule_id"), intId)...)
+	tflog.Info(ctx, "Serverbackup schedule state imported")
+}
+
+func mapResourceFields(ctx context.Context, schedule *serverbackup.BackupSchedule, model *scheduleResourceModel) error {
+	if schedule == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+	if schedule.Id == nil {
+		return fmt.Errorf("response id is nil")
+	}
+
+	model.BackupScheduleId = types.Int64PointerValue(schedule.Id)
+	idParts := []string{
+		model.ProjectId.ValueString(),
+		model.ServerId.ValueString(),
+		strconv.FormatInt(model.BackupScheduleId.ValueInt64(), 10),
+	}
+	model.ID = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+	model.Name = types.StringPointerValue(schedule.Name)
+	model.Rrule = types.StringPointerValue(schedule.Rrule)
+	model.Enabled = types.BoolPointerValue(schedule.Enabled)
+	if schedule.BackupProperties == nil {
+		model.BackupProperties = nil
+		return nil
+	}
+	ids, diags := types.ListValueFrom(ctx, types.StringType, schedule.BackupProperties.VolumeIds)
+	if diags.HasError() {
+		return fmt.Errorf("failed to map hosts: %w", core.DiagsToError(diags))
+	}
+	model.BackupProperties = &scheduleBackupPropertiesModel{
+		BackupName:      types.StringValue(*schedule.BackupProperties.Name),
+		RetentionPeriod: types.Int64Value(*schedule.BackupProperties.RetentionPeriod),
+		VolumeIds:       ids,
+	}
+	return nil
+}
+
+// If already enabled, just continues
+func enableBackupsService(ctx context.Context, model *scheduleResourceModel, client *serverbackup.APIClient) error {
+	projectId := model.ProjectId.ValueString()
+	serverId := model.ServerId.ValueString()
+	enableServicePayload := serverbackup.EnableServicePayload{}
+
+	tflog.Debug(ctx, "Enabling server backup service")
+	err := client.EnableService(ctx, projectId, serverId).EnableServicePayload(enableServicePayload).Execute()
+	if err != nil {
+		if strings.Contains(err.Error(), "Tried to activate already active service") {
+			tflog.Debug(ctx, "Service for server backup already enabled")
+			return nil
+		}
+		return fmt.Errorf("Failed to enable server backup service: %w", err)
+	}
```

Comment:
So enabling the service fails if the service is already enabled? I am asking because usually other APIs are idempotent in this case

Reply:
But if this one fails, couldn't we get the status of the service first, and only call the enable endpoint if the API is not yet enabled?

This would avoid having to rely on the error message, which can be more error prone

Reply:
Yes, enabling the service fails if already enabled. It seems this should be fixed in the service itself. This code will continue to work after it's fixed.
I think there's no API endpoint to query the status of the service, so for now it seems this error message is the only way to do it. Sorry. I don't know if we can request changing the service.